dev/planning-example.py

```diff
@@ -0,0 +1,265 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# ============================
+# Observation Planning Toolbox
+# ============================
+"""
+The observation planning toolbox consists of a number of entities that
+simplify the planning and scheduling of observations by abstracting much
+of the boilerplate code necessary to accomplish the same thing using
+core astropy or ephemerides objects.  These can be thought of as common
+idioms or convenience classes and functions that would be recreated over
+and over by astronomers and or software engineers building telescope
+scheduling software.
+
+Important assumptions:
+
+- It should be convenient to specify dates and times in the local timezone
+  of the observer.
+
+- It should be convenient to do timezone conversions easily.
+
+  This means methods that can take dates can either take astropy
+  dates (as used internally) or python datetime objects (not naive, but
+  tagged with timezone).  So dates can be passed in as timezone-aware
+  datetime objects and results can be easily converted between such.
+
+"""
+
+# ================
+# Observer objects
+# ================
+"""
+An observer defines a single terrestrial location and applicable
+environmental attributes thereof that affect observation calculations
+at that location.
+"""
+# Define an observer by full specification
+#
+# Because there may be many attributes for an observer, I suggest using
+# keyword parameters for most of them.
+from astroplan.entity import Observer
+
+obs = Observer(name='Subaru Telescope',
+               longitude='-155:28:48.900',
```

Comment:
I think in most (all?) cases we should use composition instead of inheritance.

Examples:
* An Observer has an EarthLocation member (and extra members), instead of making it an EarthLocation subclass.
* A Target has a SkyCoord member (and extra members), instead of making it a SkyCoord subclass.

@eteq, @ejeschke, all – Agreed? Or do you see a case sub-classing would be better?